### PR TITLE
Add FGM tutorial for JAX

### PR DESCRIPTION
**Pull request by npapernot**

### CODE REVIEW

**Comment by npapernot** [comment]
CLA OK to ignore

File: cleverhans/future/jax/attacks/fast_gradient_method.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	targeted=False):
+  """
+  JAX implementation of the Fast Gradient Method.
+  :param model_fn: a callable that takes an input tensor and returns the model logits.
+  :param x: input tensor.
+  :param eps: epsilon (input variation parameter); see https://arxiv.org/abs/1412.6572.
+  :param ord: Order of the norm (mimics NumPy). Possible values: np.inf, 1 or 2.
+  :param clip_min: (optional) float. Minimum float value for adversarial example components.
+  :param clip_max: (optional) float. Maximum float value for adversarial example components.
+  :param y: (optional) Tensor with true labels. If targeted is true, then provide the
```
Comment (david-berthelot) [inline cleverhans/future/jax/attacks/fast_gradient_method.py]: Let's clarify that y is probability vector, possibly one-hot for a single class target.

File: cleverhans/future/jax/attacks/fast_gradient_method.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            Targeted will instead try to move in the direction of being more like y.
+  :return: a tensor for the adversarial example
+  """
+  if y is None:
+    # Using model predictions as ground truth to avoid label leaking
+    x_labels = np.argmax(model_fn(x), 1)
+    y = one_hot(x_labels, 10)
+
+  def loss_adv(image, label):
+    pred = model_fn(np.expand_dims(image, axis=0)) #predict(params, image)
```
Comment (david-berthelot) [inline cleverhans/future/jax/attacks/fast_gradient_method.py]: Would `image[None]` also work in JAX and what interactions does it have with jit decorator.

File: tutorials/future/jax/datasets.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    urlretrieve(url, out_file)
+    print("downloaded {} to {}".format(url, _DATA))
+
+
+def _partial_flatten(x):
+  """Flatten all but the first dimension of an ndarray."""
+  return np.reshape(x, (x.shape[0], -1))
+
+
+def _one_hot(x, k, dtype=np.float32):
```
Comment (david-berthelot) [inline tutorials/future/jax/datasets.py]: May be reuse one_hot from `jax/utils` and also it could make sense to move `partial_flatten` there too.

**Pull request merged**